model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -84,6 +93,14 @@ data class PackageConfiguration(
             return false
         }
 
+        if (sourceCodeOrigin != null) {
+            return when (sourceCodeOrigin) {
+                SourceCodeOrigin.VCS -> provenance is RepositoryProvenance
+
+                SourceCodeOrigin.ARTIFACT -> provenance is ArtifactProvenance
+            }
+        }
+
```

Comment:
The below logic has to return `true` in case the package configuration does not specify a provenance. This is now possible because the check in init has been made `<=`, see above comment.